### PR TITLE
topomath eq editor updates #339, #345, #360

### PR DESCRIPTION
**Pull request by riteshsamala**

### CODE REVIEW

File: www/js/con-author.js [hunk 1]
```diff
@@ -795,6 +890,22 @@ define([
...
 						this.applyDirectives(this.authorPM.process(this.currentID, 'equation', this._model.authored.getEquation(this.currentID), true));
 				}
+				//sessionStorage contains the schema table, load options from schema table
+				var schemasList = [];
+				var schemaTabOb = JSON.parse(sessionStorage.getItem("schema_tab_topo"));
+				for(var parSchema in schemaTabOb){
+					schemaTabOb[parSchema].forEach(function(atomSchema){
+								var sname = atomSchema["Name"];
+								var obj = {value: sname, label: sname};
+								schemaWidget.addOption(obj);
```
Comment (jonwetzel) [inline www/js/con-author.js]: There's a bug with this: we add the options every time this function is called, but we never delete them.  So an extra copy of the schemas is being added every time the node editor opens.  

Perhaps adding the options should be done once when the tutor first loads (in index.php or main.js?) instead of when the node editor opens, because we don't expect the list of schemas to change during a session.

File: www/js/controller.js [hunk 2]
```diff
@@ -412,9 +415,9 @@ define([
 			}));
 
 			 //event handler for quantity node description field
```
Comment (jonwetzel) [inline www/js/controller.js]: Update comment to match code.

File: www/js/equation.js [hunk 3]
```diff
@@ -43,6 +43,8 @@ define([
 			try{
 				array.forEach(eqs, function(eq, count){
 					expressions[count] = Parser.parse(eq);
+					console.log("expr array",expressions);
```
Comment (jonwetzel) [inline www/js/equation.js]: Remove this console.log if no longer useful.

File: www/schema-table.json [hunk 4]
```diff
@@ -7,7 +7,8 @@
 				"PA": "Part A",
 				"PB": "Part B"
 			},
-			"Comments": "The whole is equal to the sum of two parts."
+			"Comments": "The whole is equal to the sum of two parts.",
+			"slot names": "Whole, Part A, Part B"
```
Comment (jonwetzel) [inline www/schema-table.json]: As previously discussed, the slot names field is redundant with the Mapping.  Go ahead and remove all "slot names" fields.

**Comment by riteshsamala** [comment]
I have updated the code in latest commit on current branch. It also includes enhancements. I also found one bug while testing, entity text box was accepting empty entity input value before semicolon like ";entity1". Fixed it as well.
